Solutions/moritz_w/Lesson_4/eventlog.py

```diff
@@ -0,0 +1,88 @@
+"""
+Windows Event Log Parser
+Author: Moritz Wiesinger
+"""
+
+from datetime import datetime
+import unittest
+
+
+class LogEntry:
+    def __init__(self):
+        self.log_name = None
+        self.source = None
+        self.date = None
+        self.event_id = None
+        self.task = None
+        self.level = None
+        self.opcode = None
+        self.keyword = None
+        self.user = None
+        self.user_name = None
+        self.computer = None
+        self.description = []
+
+
+def parser(content):
+    """
+    Parses the content of a log.
+
+    :param content: List of event log lines
+    :type content: list[str]
+    :return: list[LogEntry]
+    """
+    ret = []
+    current_log_entry = None
+    for i in range(len(content)):
+        line = content[i]
+
+        if "Event[" in line:
+            # new event
+            last_log_entry = current_log_entry
+            current_log_entry = LogEntry()
+
+            if last_log_entry is not None:
+                ret.append(last_log_entry)
+                last_log_entry.description.pop()
+        else:
+            if ': ' in line:
+                data = line.split(': ')[1]
+                data = data[0:len(data) - 1]
```

Comment:
Simpler way is: data = data[:-1]
You can index from the end of an array using negative indexes.